### PR TITLE
Consultation request pages

### LINKED ISSUE #13: Consultation Request pages
- List page (maybe with filters), buttons for supporting it and creating consultation from it
- no need for a details page
- create/edit page (or just a modal) - the subject selector of the consultation form should be used, might need some refactor

### PR DESCRIPTION
**Pull request by kisbogdan-kolos**
closes #13

### CODE REVIEW

File: src/pages/requests/RequestListPage.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  if (!loggedInUser && !loggedInUserLoading) {
+    return <ErrorPage status={401} />
+  }
+
+  return (
+    <>
+      <Helmet title="Konzultáció kérések" />
+      <Heading size="xl" textAlign="center" mb={3}>
+        Konzultáció kérések
+      </Heading>
```
Comment (DannySS123) [inline src/pages/requests/RequestListPage.tsx:44]: PageHeading

File: src/pages/requests/RequestDetailsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+  const isOwner = request.initializer.id === loggedInUser.id
+  const isSupporter = request.supporters.some((s) => s.id === loggedInUser.id)
+
+  return (
+    <>
+      <Helmet title={request.name} />
+      <Heading textAlign="center" mb={3}>
+        {request.name}
+      </Heading>
```
Comment (DannySS123) [inline src/pages/requests/RequestDetailsPage.tsx:66]: PageHeading

File: src/components/commons/MajorAvatar.tsx [hunk 3]
```diff
@@ -5,15 +5,13 @@ import { MajorIcon } from './MajorIcon'
...
-  return (
-    <Avatar
-      size="md"
-      bg={subject.majors.length > 1 ? 'orange.500' : majorColorForIcon[subject.majors[0]]}
-      icon={<MajorIcon majors={subject.majors} />}
-      src={''}
-    />
-  )
+export const MajorAvatar = ({ subject, monochrome = false }: Props) => {
+  let bgcolor = subject.majors.length > 1 ? 'purple.500' : majorColorForIcon[subject.majors[0]]
```
Comment (DannySS123) [inline src/components/commons/MajorAvatar.tsx:12]: Multisubject color should stay orange
But feel free to experiment and find better colors for the subjects in a different pr :D

**Review by DannySS123 — CHANGES_REQUESTED** [review]
Very nice job! 🥇 

Some problems:
- Konzi Request Törlés gomb does nothing
- too long name goes outside of the box, do what I did on the other pages
image
- there is a lot of empty space in a card:
image
maybe consider doing the 2 column grid as on the groupspage?
- there should be a "Megtartom" button on the RequestDetails page, that goes to a new consultation page with filled out data. This button can alse be on the cards in the requests list, than there won't be that much space and we won't need 2 columns.
- the new consultation page should have a field for "Megvalósított kérés" (perhaps a better name). If it's selected the data should be filled. 
There are many ways to do this, for example with a checkbox for completing a request and if it's checked and a request is selected, the filled out data (subject, etc..) should be disabled for editing, or enabled but then changing it will uncheck the checkbox and remove the selected request. These are just some ideas, experiment a bit and find a nice solution that is confortable to use.

This might seem a lot, but altogether, great work, you can be proud! We are too ❤️ 
If I left out anything @Tschonti will probobly mention it later xd

**Comment by Tschonti** [comment]
I think the Konzi megtartása stuff can be a separate task for tomorrow. If you're interested, we can do it together. But this PR is pretty large already, those features should be handled separately (also, there a few additional stuff for those)

**Comment by kisbogdan-kolos** [comment]
> I think the Konzi megtartása stuff can be a separate task for tomorrow. If you're interested, we can do it together. But this PR is pretty large already, those features should be handled separately (also, there a few additional stuff for those)

Okay. The 'konzi megtartása' button should be easy, I will include it here, and leave the other stuff for a different PR.

**Review by DannySS123 — APPROVED** [review]
Seems good 👌

**Review by Tschonti — APPROVED** [review]
Very nice job! I edited the layout of the buttons on the request list. Previously:
image
Now:
image
(unchanged on smaller devices)

One idea: instead of showing the expiry date on the request list, we could just show how many days are left (or months, if it's far away). Just to make sure people don't confuse requests with konzis.

File: src/pages/requests/RequestDetailsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+      )}
+      {request.consultations.length > 0 && (
+        <>
+          <Heading size="lg" mb={2}>
+            Konzultációk ({request.consultations.length})
+          </Heading>
+          <VStack alignItems="stretch" mb={6}>
+            {isLoading ? (
+              <LoadingConsultationList />
+            ) : (
```
[the remark this answers is not in the document]
Comment (kisbogdan-kolos) [reply src/pages/requests/RequestDetailsPage.tsx:146]: I copied this from the consultations list and forgot to delete it.

**Pull request merged**